pkg/controller/node/nodecontroller.go

```diff
@@ -203,6 +211,31 @@ func (nc *NodeController) getCondition(status *api.NodeStatus, conditionType api
 // not reachable for a long period of time.
 func (nc *NodeController) monitorNodeStatus() error {
 	nodes, err := nc.kubeClient.Nodes().List(labels.Everything(), fields.Everything())
+	oldSize := len(nc.knownNodeSet)
+	observedSet := make(util.StringSet)
+	addedNodes := 0
+	for _, node := range nodes.Items {
+		observedSet.Insert(node.Name)
+		if !nc.knownNodeSet.Has(node.Name) {
+			glog.V(1).Infof("NodeController observed a new Node: %#v", node)
+			nc.recordNodeEvent(node.Name, fmt.Sprintf("Registered Node %v in NodeController", node.Name))
+			nc.knownNodeSet.Insert(node.Name)
```

Comment:
It's for quick check if we deleted something or not. Difference of sets is slower. If you already mentioned it, I'd rather move construction of observedSet into the second if, as it'll be rather rare (hopefully)